game/src/ecs/systems/HealthSystem.java

```diff
@@ -0,0 +1,64 @@
+package ecs.systems;
+
+import ecs.components.AnimationComponent;
+import ecs.components.HealthComponent;
+import ecs.components.MissingComponentException;
+import ecs.damage.Damage;
+import ecs.entities.Entity;
+import java.util.List;
+import mydungeon.ECS;
+
+/**
+ * The HealthSystem offsets the damage to be done to all entities with the HealthComponent. Triggers
+ * the death of an entity when the hitpoints have fallen below 0.
+ */
+public class HealthSystem extends ECS_System {
+    @Override
+    public void update() {
+        for (Entity entity : ECS.entities) {
+            entity.getComponent(HealthComponent.name)
+                    .ifPresent(
+                            component -> {
+                                {
+                                    final AnimationComponent ac =
+                                            (AnimationComponent)
+                                                    entity.getComponent(AnimationComponent.name)
+                                                            .orElseThrow(
+                                                                    () ->
+                                                                            new MissingComponentException(
+                                                                                    AnimationComponent
+                                                                                            .name));
+
+                                    HealthComponent hpComponent = (HealthComponent) component;
+
+                                    // is the entity dead?
+                                    if (hpComponent.getCurrentHitPoints() <= 0) {
+                                        hpComponent.triggerOnDeath();
+                                        ac.setCurrentAnimation(hpComponent.getDieAnimation());
+                                        // todo: After animation is finished
+                                        ECS.entitiesToRemove.add(entity);
+                                    } else {
+                                        // make damage
+                                        List<Damage> damageToGet = hpComponent.getDamageList();
+                                        int dmgAmmount = 0;
+                                        for (Damage dmg : damageToGet) {
+                                            // place to increase or decrease dmg based on skills,
```

Comment:
Was will dieser Kommentar sagen? Liest der nicht einfach nur den Code vor?

Reply:
Äh ne? 
Das ist der Ort, an dem wir (wenn wir das haben) Rüstungswerte und Co verrechnen. Darum Brauch ich auch die Schleife

Reply:
Genau. Ich sehe im Code eine Schleife, die Damage-Werte zusammenzählt. Und genau das sagt der Kommentar auch. Für mich liest der Kommentar einfach nur den Code vor und erklärt mir nicht, warum hier etwas passiert. Gehört in die Kategorie "kann weg".

Reply:
Da steht: 
 ` // place to increase or decrease dmg based on skills,  items`
im Code ist aber nix von Skills, Items zu sehen. 
Also kann das Kommentar den Code gar nicht vorlesen. 

Ich kann aber nix dagegen sagen, dass dich der Kommentar verwirrt.

Reply:
Haste wahr, im Code ist nix von Skills zu sehen - das geht einfach nur allgemein über Damage-Objekte aus der `hpComponent`. Aber dann ist der Kommentar für mich noch verwirrender :)